projects/opendr_ws_2/src/opendr_perception/opendr_perception/object_tracking_3d_ab3dmot_node.py

```diff
@@ -78,19 +88,26 @@ def callback(self, data):
         :param data: input message
         :type data: sensor_msgs.msg.Image
         """
-
+        if self.performance_publisher:
+            start_time = perf_counter()
         # Convert sensor_msgs.msg.Image into OpenDR Image
         point_cloud = self.bridge.from_ros_point_cloud(data)
         detection_boxes = self.detector.infer(point_cloud)
+        tracking_boxes = self.learner.infer(detection_boxes)
```

Comment:
Would it maybe make sense here to execute it only when needed?
```suggestion
        if self.tracking_id_publisher or self.performance_publisher:
            tracking_boxes = self.learner.infer(detection_boxes)
```

Reply:
Thank you for the review Stefania! 

The reasoning behind this was that this node is supposed to do tracking so it is a bit counterintuitive to disable it. It runs the Voxel3D object detection tool, so one should use that tool's node instead of this one. Nevertheless i can imagine someone wanting to have both options available when using this so i will add your suggestion and apply it to the ROS1 node too.

Reply:
Ok, thank you for the explanation!
I just checked the code and saw that it was used only if `self.tracking_id_publisher` was set.
But for me it is also fine to keep it without the if condition.

Reply:
Yeah i think code-wise it makes more sense with the if :smile: Thanks!